File: window/headerwindow.py
from PyQt5.QtCore import QCoreApplication, pyqtSignal
from window.template import OpaqueWindow

class HeaderWindow(OpaqueWindow):
  onMoved = pyqtSignal(int, int)

  def __init__(self, engine, **kwargs):
    super().__init__(engine, 'qml/Header.qml', **kwargs)
    self.rootObject().drag.connect(self.drag)
    self.rootObject().exit.connect(lambda: QCoreApplication.instance().exit())

  def drag(self, x, y):
    self.move(self.x() + x, self.y() + y)
    self.onMoved.emit(self.x(), self.y())

File: app/src/window/headerwindow.cpp
#include "headerwindow.h"

HeaderWindow::HeaderWindow(QQmlEngine* engine)
    : Window(engine, false)
{
    setSource(QUrl("qrc:/ui/Header.qml"));

    connect(global(), SIGNAL(compassVisibleChanged()),
        this, SLOT(onCompassVisibleChanged()));
    connect(rootObject(), SIGNAL(drag(int, int)),
        this, SLOT(onDrag(int, int)));
    connect(rootObject(), SIGNAL(exit()),
        QCoreApplication::instance(), SLOT(quit()));
}

void HeaderWindow::onCompassVisibleChanged()
{
    bool visible = global()->property("compassVisible").toBool();
    setVisible(visible);
}

void HeaderWindow::onDrag(int dx, int dy)
{
    move(x() + dx, y() + dy);
    emit moved(x(), y());
}
